### ISSUE TITLE
Anemone 3DS crashing upon startup

### ISSUE THREAD

**Issue by Ludicolo2799** [astronautlevel2/Anemone3DS#228]
I had a custom Erased theme downloaded from ThemePlaza installed, and decided I wanted to change it. When I booted up Anemone however, I got this error message and stack dump. Any suggestions on how to resolve the issue? 
![15940750267503898603612339613193](https://user-images.githubusercontent.com/67931288/86659294-5ab1b880-bfaf-11ea-88d4-1737e825d8c6.jpg)

**Comment by Ludicolo2799** [comment]
Navigating to Nintendo 3DS -> 32 character ID -> 32 character ID -> extdata -> 00000000 -> and then deleting 0000008f did not resolve issue. Tried reinstalling Anemone via QR code through FBI, issue did not resolve.

**Comment by Helloman892** [comment]
So in answer to your question, this already feels like a very fringe edge case - the crash isn't happening in our code, it's happening pretty low to the ground, so it's a hard one to pinpoint. Definitely looks like something wacky is going on, though.

@astronautlevel2, thoughts?

**Comment by EdEddED** [comment]
Not sure if anyone is looking into this still, but I'm going to bump and say this is not an isolated incident. I've just had my New 3DS XL Act up in this exact manner. Let me explain what happened prior to this problem happening, and include a dump file or two to maybe help out.

The problem started when I attempted a shuffle install of several themes at a time. Before this point I have been having audio glitches with theme music at random when leaving certain games and apps for the home menu. After attempting the shuffle install of several themes, I restarted my 3DS to find everything working as normal. The problem began when it came time to shuffle onto the next theme - The audio became garbled static and I powered off the 3DS for the night. When I came back to fix it in the morning, the 3DS was soft-bricked, and would crash every time the home menu was attempted to be loaded. I do have a dump log of that first event happening. 

Removing the microSD card worked, so I cleared the Home Menu extdata on the card to reset to the default theme. But now I have a new problem - specifically the problem @Ludicolo2799 originally reported. The rest of the 3DS is now working like normal, but Anemone 3DS will no longer boot at all - it loads the themes and splashes but it crashes before you get to the menu. 

I've attached the two dump files from luma3DS and placed them in a zip file. 
[ARM9 Crash Dumps.zip](https://github.com/astronautlevel2/Anemone3DS/files/4976114/ARM9.Crash.Dumps.zip)
The first one [...00.dmp] may not be of too much use since it was when the Home menu was broken, but the second one [...01.dmp] may be a bit more useful.

**Comment by astronautlevel2** [comment]
Nothing Anemone3DS does should cause an Arm9 error, this feels like some sort of SD corruption issue. 

Anemone3DS and the Official Theme Manager on the home menu install themes in different ways, Anemone takes advantage of the theme cache to install themes.

Some troubleshooting you can do is remove all of the artifacts Anemone3DS puts on your console - most importantly this means clearing the theme extdata and home menu extdata. The Anemone3DS includes instructions on clearing theme extdata, but you should also try clearing the home menu extdata like so:

```
1. Navigate to the following folder on your SD card: /Nintendo 3DS/(32 Character ID)/(32 Character ID)/extdata/00000000/
2. Delete the corresponding folder for your region:
USA: 0000008f
EUR: 00000098
JPN: 00000082
KOR: 000000A9
```

Additionally, you should try backing up your /themes/ folder and creating a new, empty folder. If Anemone is still crashing, try to install it on a different SD card and try again.

**Comment by EdEddED** [comment]
I've tried clearing all the theme data on my old SD card, and it was still causing problems, so I've moved to a new microSD card, and it's working again. As I did so, I discovered that the SD Card's file system was indeed corrupted, so I had to use chkdsk on Windows to fix it [because apparently the 3DS doesn't compensate for errors in the filesystem???], then copy everything that was there to a new card. The old card still seems to be working fine on a hardware level, but I think it's safe to say I'm not trusting that card anymore.

**Comment by Ludicolo2799** [comment]
I'm going to buy a new SD card with my next paycheck, switch over to it, and see if that fixes my issue. Did anyone format an SD to fit your system through GodMode? I have a 64 GB card, formatted to Fat32, and I think that may be my issue.On Jul 27, 2020 12:12 PM, EdEddED <notifications@github.com> wrote:
I've tried clearing all the theme data on my old SD card, and it was still causing problems, so I've moved to a new microSD card, and it's working again. As I did so, I discovered that the SD Card's file system was indeed corrupted, so I had to use chkdsk on Windows to fix it [because apparently the 3DS doesn't compensate for errors in the filesystem???], then copy everything that was there to a new card. The old card still seems to be working fine on a hardware level, but I think it's safe to say I'm not trusting that card anymore.

**Comment by navarayuk** [comment]
As much as this could be a SD card problem I don't believe so, since 2.0.1 was running fine and I found you had a update I installed 2.1.0 and then this happened to me also I have a Class 10 SandiskU SDXC 1 this card is new and runs everything else 100% fine part from unable to now change themes, there might be something corrupting but it seems to be only if this app is updated, I also noted when I have the problem it's when I have a good few themes I removed most of them and it comes back to life, so it might also be something to do with how many themes it can read.

**Comment by astronautlevel2** [comment]
@navarayuk I believe your issue is a different one from the one mentioned in this thread - I think it should've been fixed by #239 

Closing this issue to prevent people from attributing unrelated issues to it.
